[DEVDOCS-5787]: [New] GQL Storefront Orders API, Create new doc (#290)
bigcommerce/docs: docs/api-docs/storefront/graphql/graphql-api-orders.mdx

Reviewer: currently we don't have mutations for orders.

Author: Doh! Thanks for catching that.

Reviewer: I'd suggest removing any of my test data and leveraging a common customer / customer data we use in other doc examples.

Reviewer: Same here.   If we have common examples we use in product catalog, products, variants I would update the example so it is consistent.   Other than that, it reads really well.

# GraphQL Storefront API: Orders (beta)

<Callout type="info">
    This feature is currently in beta and the following operation is only available to participants. 
</Callout>

## Overview

Access order information from your BigCommerce store using the GraphQL Storefront schemas. You can make requests to the BigCommerce GraphQL service using the `site` node.

Use both query and mutation operation types to fetch and write data using the BigCommerce graph. Create arguments for fields, like filtering `orders` by `entity_Id`. You can learn more example site operations by reading on. 

### Query site orders detail

Use the `orders` object type to fetch data to display orders history and details on the storefront graph using the `site` node. 

> Use the `GetAllOrders` field query to fetch all orders for a specific customer. You can get a customer's order history, and get details like:

Reviewer 2: ```suggestion
Use the `GetAllOrders` query field to fetch all orders for a specific customer. You can get a customer's order history, and get details like:
```

</Callout>

## Query site order details

Use the `order` object type to get detailed information on the storefront graph using the `site` node.

Use the `GetOrderDetails` field query to fetch order details like:

- Specific customer orders
- Specific order data filtered by date, or status

Note: Cursor pagination is supported to refine fetched information.

## Tokens

To make requests, create a [store-level](/docs/start/authentication/api-accounts#store-level-api-accounts) or [app-level](/docs/start/authentication/api-accounts#app-level-api-accounts) API account with one or more of the following token creation OAuth scopes:

| UI Name | Permission | Parameter | Description |
| :--- | :--- | :--- | :--- |
| storefront API tokens | modify | `store_storefront-api` | Create payment access tokens, process payments |
| storefront API customer impersonation tokens | modify | `store_storefront_api_customer_impersonation` | Create GraphQL storefront API bearer tokens that allow customer impersonation |

No additional scopes are required to use the GraphQL storefront API. For more information, see [token creation OAuth scopes](/docs/start/authentication/api-accounts#token-creation-scopes).

> To generate a bearer token your application can pass to authenticate calls to the GraphQL storefront API, using either the [Create a storefront token](/docs/rest-authentication/tokens#create-a-token) or the [Create a customer impersonation token](/docs/rest-authentication/tokens/customer-impersonation-token#create-a-token) REST endpoint. On a Stencil storefront, you can also access a token directly from the page context. Learn more about [Using auto-generated tokens in Stencil themes](/docs/storefront/graphql#using-auto-generated-tokens-in-stencil-themes).

Reviewer 2: ```suggestion
To generate a bearer token your application can pass to authenticate calls to the GraphQL Storefront API, using either the [Create a storefront token](/docs/rest-authentication/tokens#create-a-token) or the [Create a customer impersonation token](/docs/rest-authentication/tokens/customer-impersonation-token#create-a-token) REST endpoint. On a Stencil storefront, you can also access a token directly from the page context. Learn more about [Using auto-generated tokens in Stencil themes](/docs/storefront/graphql#using-auto-generated-tokens-in-stencil-themes).
```
Not sure if we were still planning on keeping it caps in the style guide

# GraphQL Storefront API: Orders (beta)

<Callout type="info">
    This feature is currently in beta and is only available to participants. 
</Callout>

## Overview

BigCommerce's GraphQL Storefront API Orders provides the same access to order objects as the REST Storefront API does. Developers can use frontend GraphQL to create an end-to-end shopper experience. 

This GraphQL API endpoint powers customer solutions as shoppers evolve from guests, to registered customers. Orders related mutations and queries can do the following:

> ### Get order and details

Reviewer: ```suggestion
### Get order details for a specific order
```

> Get all order details for a specific customer using the `GetOrderDetails` query.

Reviewer: ```suggestion
Get specific details for a specific order using the `order` query on the storefront graph under the site node.
```

> Use `GetOrderDetails` to:

Reviewer: ```suggestion
Use `Order` to:
```

I don't love just calling out the specific node with out giving some sort of path.  Something like site>order or site>orders but I guess that can be called out in the examples provided.  I'm 100% open to suggestions but this caused friction for me when I first started reading our graph docs.

Author: Great feedback. The docs team is re-thinking the way we document our graphs. This helps shape some of that re-work.

Author: I changed up the intro to better introduce the operation.

>   - Query for a specific order 

Reviewer: These are mixed up.  

We need to switch these since "orders" allows you to filter by status, date, and offers pagination along with specifics you might want such as 

"order" allows you to query the specific details.

Author: Switching it up with better headers.

  - Filter orders by order status
  - Filter orders by order date
  - Support for cursor pagination

### Get all orders and details

> Get all orders for a specific customer using the `GetAllOrders` query.

Reviewer: ```suggestion
Get all orders for a specific customer using the `Orders` query.   This can be found under the storefront graph site node.
```

Use the `GetAllOrders` query to get details like:

> - Billing address, including the customer's information

Reviewer: This is where we need to describe getting all orders by status, date as well as pagination.
